ios/RNSScreenWindowTraits.mm

```diff
@@ -22,43 +22,18 @@ + (void)assertViewControllerBasedStatusBarAppearenceSet
 
 + (void)updateStatusBarAppearance
 {
-#if !TARGET_OS_TV
+#if !TARGET_OS_TV && !TARGET_OS_VISION
   [UIView animateWithDuration:0.4
                    animations:^{ // duration based on "Programming iOS 13" p. 311 implementation
-#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_13_0) && \
-    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0
-                     if (@available(iOS 13, *)) {
-                       UIWindow *firstWindow = [[[UIApplication sharedApplication] windows] firstObject];
-                       if (firstWindow != nil) {
-                         [[firstWindow rootViewController] setNeedsStatusBarAppearanceUpdate];
-                       }
-                     } else
-#endif
-                     {
-                       [UIApplication.sharedApplication.keyWindow.rootViewController setNeedsStatusBarAppearanceUpdate];
-                     }
-                   }];
+    [RCTKeyWindow().rootViewController setNeedsStatusBarAppearanceUpdate];
```

Comment:
Small note on why I changed it: 

`RCTKeyWindow()` internally loops over windows: 

```objc
for (UIWindow *window in RCTSharedApplication().windows) {
    if (window.keyWindow) {
      return window;
    }
  }
```

The `window` property is available from iOS 2.0 so I think that this check wasn't needed here and we are safe to just use `RTCKeyWindow()`